src/parser.js

```diff
@@ -13,91 +25,50 @@ type CSSDecl = {
 
 export function parseCSS (
   css: string,
-  options: {
-    nested?: boolean,
-    inlineMode: boolean,
-    matches: number,
-    name: string,
-    hash: string,
-    canCompose?: boolean
-  } = {
-    nested: true,
-    inlineMode: true,
-    matches: 0,
-    name: 'name',
-    hash: 'hash',
-    canCompose: false
-  }
+  extractStatic: boolean
 ): {
-  rules: string[],
-  hasOtherMatch: boolean,
-  hasVar: boolean,
-  composes: number,
-  hasCssFunction: boolean
+  staticCSSRules: Array<string>,
+  styles: { [string]: any },
+  composesCount: number
 } {
   // todo - handle errors
-  const root = parse(css)
-  if (options.nested !== false) postcssNested(root)
-
+  let root = parse(css)
   let vars = 0
   let composes: number = 0
-  let hasCssFunction = false
-  root.walkDecls((decl: CSSDecl): void => {
-    if (decl.prop === 'name') decl.remove()
-    if (decl.value.match(/attr/)) {
-      hasCssFunction = true
+
+  postcssNested(root)
+
+  root.walkRules((rule: Rule) => {
+    // TODO: do this everywhere except `,xxx9xxx`
+    if (/\bxxx\d+xxx/.exec(rule.selector)) {
+      rule.selector = `.${rule.selector}`
```

Comment:
I don't think it's a good idea to add the `.`, I think it makes more sense that when you interpolate a class, you have to add the `.`. This would also stop people from interpolating ids and other selectors.